internal/controller/authenticator/jwtcachefiller/jwtcachefiller_test.go

```diff
@@ -363,41 +712,217 @@ func TestController(t *testing.T) {
+					},
+					Spec: *badIssuerJWKSURISchemeJWTAuthenticatorSpec,
+				},
+			},
+			syncKey: controllerlib.Key{Name: "test-name"},
+			wantStatusConditions: status.ReplaceConditions(
+				allHappyConditionsSuccess(goodIssuer, frozenMetav1Now, 0),
+				[]metav1.Condition{
+					happyIssuerURLValid(frozenMetav1Now, 0),
+					sadReadyCondition(frozenMetav1Now, 0),
+					unknownAuthenticatorValid(frozenMetav1Now, 0),
+					sadJWKSURLValidScheme("http://.café.com/café/café/café/coffee/jwks.json", frozenMetav1Now, 0),
+				},
+			),
+			wantStatusPhase: "Error",
+			wantSyncLoopErr: testutil.WantExactErrorString("jwks_uri http://.café.com/café/café/café/coffee/jwks.json has invalid scheme, require 'https'"),
 		},
+		// cannot be tested the way we are invoking oidc.New as we don't provide enough configuration
+		// knobs to actually invoke the code in a broken way.  We always give a good client, good keys, and
+		// good signing algos.  In the future if we allow any of these to be configured we may have opportunity
+		// to test for errors.
+		// {name: "newCachedJWTAuthenticator: could not initialize oidc authenticator..." },
 	}
 
 	for _, tt := range tests {
 		tt := tt
 		t.Run(tt.name, func(t *testing.T) {
 			t.Parallel()
 
-			fakeClient := pinnipedfake.NewSimpleClientset(tt.jwtAuthenticators...)
-			informers := pinnipedinformers.NewSharedInformerFactory(fakeClient, 0)
+			pinnipedAPIClient := pinnipedfake.NewSimpleClientset(tt.jwtAuthenticators...)
+			pinnipedInformers := pinnipedinformers.NewSharedInformerFactory(pinnipedAPIClient, 0)
 			cache := authncache.New()
-			testLog := testlogger.NewLegacy(t) //nolint:staticcheck  // old test with lots of log statements
+
+			var log bytes.Buffer
+			logger := plog.TestLogger(t, &log)
 
 			if tt.cache != nil {
 				tt.cache(t, cache, tt.wantClose)
 			}
 
-			controller := New(cache, informers.Authentication().V1alpha1().JWTAuthenticators(), testLog.Logger)
+			controller := New(
+				cache,
+				pinnipedAPIClient,
+				pinnipedInformers.Authentication().V1alpha1().JWTAuthenticators(),
+				frozenClock,
+				logger)
 
 			ctx, cancel := context.WithCancel(context.Background())
 			defer cancel()
 
-			informers.Start(ctx.Done())
+			pinnipedInformers.Start(ctx.Done())
 			controllerlib.TestRunSynchronously(t, controller)
 
 			syncCtx := controllerlib.Context{Context: ctx, Key: tt.syncKey}
 
-			if err := controllerlib.TestSync(t, controller, syncCtx); tt.wantErr != nil {
-				testutil.RequireErrorStringFromErr(t, err, tt.wantErr)
+			if err := controllerlib.TestSync(t, controller, syncCtx); tt.wantSyncLoopErr != nil {
+				testutil.RequireErrorStringFromErr(t, err, tt.wantSyncLoopErr)
 			} else {
 				require.NoError(t, err)
 			}
-			require.Equal(t, tt.wantLogs, testLog.Lines())
-			require.Equal(t, tt.wantCacheEntries, len(cache.Keys()))
+
+			actualLogLines := logLines(log.String())
+			require.Equal(t, len(actualLogLines), len(tt.wantLogs), "log line count should be correct")
```

Comment:
Expected and actual are passed in the wrong order.

Reply:
Also true in lots of other usages of `require.Equal` in this file.

Reply:
fixed in next push.